pkg/client/addon_deployment_update.go

```diff
@@ -13,18 +13,18 @@ func (h *V1Client) UpdateAddonDeployment(cluster *models.V1SpectroCluster, body
 	uid := cluster.Metadata.UID
 
 	// check if profile id is the same - update, otherwise, delete and create
-	is, replaceUID := isProfileAttachedByName(cluster, newProfile)
+	is, replaceUID := IsProfileAttachedByName(cluster, newProfile)
 	if is {
 		body.Profiles[0].ReplaceWithProfile = replaceUID
 	}
 
 	resolveNotification := true
 	params := clusterC.NewV1SpectroClustersPatchProfilesParamsWithContext(h.Ctx).WithUID(uid).WithBody(body).WithResolveNotification(&resolveNotification)
-	err := patchWithRetry(h, params)
+	err := PatchWithRetry(h, params)
 	return err
 }
 
-func isProfileAttachedByName(cluster *models.V1SpectroCluster, newProfile *models.V1ClusterProfile) (bool, string) {
+func IsProfileAttachedByName(cluster *models.V1SpectroCluster, newProfile *models.V1ClusterProfile) (bool, string) {
```

Comment:
any particular reason to export this?

Reply:
this is so unit tests can be in a different module.